### DISCUSSION TITLE
An official instance.

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [searxng/searxng#2582] [Q&A] [answered]
I want to use SearXNG due to its evident superiority to most other SEs in that it can aggregate results from multiple. However, https://searx.space/# shows quite a multitude of instances to choose from. Consequently, are any instances hosted (officially) by the developmental team?

If one doesn't exist, I believe that it should. I'm not the sole person to think that an official instance should exist, per https://www.reddit.com/r/privacytoolsIO/comments/h80oum/comment/g6wz1hr/?utm_source=share&utm_medium=web2x&context=3

> Would be neat if searx hosted an official server that was funded through donations or something.

Bitwarden generates quite a significant revenue stream for its project by providing an official instance, since it will always serve the latest version, and have the best uptime, fundamentally.

**Comment by unixfox** [comment, marked as answer]
Searxng maintainers prefer dedicate their precious free time to improve the software instead of spending the time to maintain an official instance.

So no there will never be an official instance for searxng.

**Reply by roke-julian-lockhart** [reply]
> [Would be neat if searx hosted an official server that was funded through donations or something.](https://www.reddit.com/r/privacytoolsIO/comments/h80oum/comment/g6wz1hr/?utm_source=share&utm_medium=web2x&context=3)

If the development team were paid for it, it could hire developers dedicated to the task of maintaining the instance rather than divert existent members' attention to it, so I don't understand how that's a reason.

**Reply by unixfox** [reply]
> > [Would be neat if searx hosted an official server that was funded through donations or something.](https://www.reddit.com/r/privacytoolsIO/comments/h80oum/comment/g6wz1hr/?utm_source=share&utm_medium=web2x&context=3)
> 
> If the development team were paid for it, it could hire developers dedicated to the task of maintaining the instance rather than divert existent members' attention to it, so I don't understand how that's a reason.

Are you being serious? Paying someone for maintaining an official instance would cost us a lot.

An open source project do not receive enough funds for paying even someone part time on working on maintaining an official instance.

We don't even get enough funds to pay ourselves if we wanted to work part time on searxng!

Money doesn't grow on trees. We did open to donations a few months ago and we were receiving like only 100€/month max.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/searxng/searxng/discussions/2582#discussioncomment-6605326

Bitwarden, as an example, gains more than enough money to host their official password management instance (https://vault.bitwarden.com) by locking certain features behind a paywall if they necessitate hosted servers. For instance, like how Bitwarden locks its TOTP generation features, you could provide settings sync features for the user via an account system, thus allowing them to not have to backup their configuration in case the user's site cookies are deleted.

I understand that donations indeed rarely are adequate as a source of income. Consequently, I hope you realize now that I am not suggesting such a thing. Thanks.

**Comment by return42** [comment]
> since it will always serve the latest version, and have the best uptime

https://github.com/searxng/searx-instances is the project where the maintenance of the instance list of the searx.space is done.  Uptime of SearXNG instances is monitored and the rules to have a instance listed on https://searx.space are very strict -->
https://github.com/searxng/searx-instances/blob/master/.github/ISSUE_TEMPLATE/add-instance.yaml  // By example uptime at **minimum 90%**

-----

Related: 

- https://github.com/searxng/searx-instances/issues/330

**Reply by roke-julian-lockhart** [reply]
But I have no idea how to choose which to use. Is there any way of checking how many users an instance has?

**Reply by unixfox** [reply]
No, ideally choose one that has a good uptime, is located close to you (country) and have a low search response time.

**Reply by return42** [reply]
> But I have no idea how to choose which to use.

randomly? At the end you have to trust anyone from the list or you have to set up one instance by your own.

> Is there any way of checking how many users an instance has?

No .. we do not track users ;-)

**Reply by roke-julian-lockhart** [reply]
@return42, really not even usage metrics? That seems rather like an oversight.

**Reply by return42** [reply]
@rokejulianlockhart we do not store any data point of the users. We do not have sessions / SearXNG works stateless and you can also use SearXNG without JS. We invest a lot of time to keep it that way. Even the limiter stores the IPs only for a short time and anonymized.

The only thing we record is the quality of the engines --> https://darmarit.org/searx/stats

**Reply by return42** [reply]
OT @rokejulianlockhart you should fix you github homepage https://github.com/rokejulianlockhart .. we all can read your access to token ;-)

**Reply by roke-julian-lockhart** [reply]
> https://github.com/searxng/searxng/discussions/2582#discussioncomment-6605098

@return42, why not use JS? I've never heard of that being problematic. In fact, some sites seem to be entirely JS nowadays.

> https://github.com/searxng/searxng/discussions/2582#discussioncomment-6605160

@return42, it's a public access token. It only grants reads. Means that I don't have to synchronize bios between my 1,650 accounts XD
